catwalk/models/metaicl.py

```diff
@@ -0,0 +1,133 @@
+    """A model that reproduces the formatting and truncation of MetaICL."""
+
+    def __init__(self, pretrained_model_name_or_path: str, max_length_per_example : int = 256, continuation_seperator: str = '\n', example_seperatior='\n\n\n'):
+        super().__init__(pretrained_model_name_or_path)
+        self.max_length_per_example = max_length_per_example
+        self.continuation_seperator = continuation_seperator
+        self.example_seperatior = example_seperatior
+        self.instances_truncated = 0
+        self.instances_total = 0
+
+
+    def predict_chunk(
+        self,
+        task: Task,
+        instances: Sequence[Dict[str, Any]],
+        model: _Model,
+        tokenizer: _Tokenizer,
+        batch_size: int = 32,
+        num_shots: int = 0,
+        fewshot_seed: int = None
+    ) -> Iterator[Dict[str, Any]]:
+        instance_index_to_tuple_indices: Mapping[int, List[int]] = collections.defaultdict(list)
+        tuples: List[Tuple[str, str]] = []
+        self.instances_truncated = 0
+        self.instances_total = 0
+
+        rc_instances: List[RankClassificationInstance] = []
+        for i, instance in enumerate(instances):
+            # pre-instance processsing (both fewshot and test example)
+            instance['input'] = self._apply_meta_icl_per_instance_truncation(instance, tokenizer, is_icl_demonstration=False)
+            fewshot_instances=task.get_fewshot_instances(num_shots, random_seed=fewshot_seed if fewshot_seed is not None else i, exceptions=instance)
+            truncated_fewshot_instances = []
+            for i, fewshot_instance in enumerate(fewshot_instances):
+                if i == 0:
+                    fewshot_instance['input'] = self._apply_meta_icl_per_instance_truncation(fewshot_instance, tokenizer, is_first=True)
+                else:
+                    fewshot_instance['input'] =  self._apply_meta_icl_per_instance_truncation(fewshot_instance, tokenizer)
+                truncated_fewshot_instances.append(fewshot_instance)
+            
+            rc_instances.append(task.convert_instance(
+                instance,
+                InstanceFormat.RANK_CLASSIFICATION,
+                fewshot_instances=truncated_fewshot_instances,
+                continuation_seperator=self.continuation_seperator,
+                example_seperator=self.example_seperatior,
+            ))
+        
+        rc_instances = [self._apply_meta_icl_concatinated_input_truncation(instance, tokenizer) for instance in rc_instances]
+
+        # get all the tuples
+        for instance_index, rc_instance in enumerate(rc_instances):
+            for instance_request in rc_instance.choices:
+                instance_index_to_tuple_indices[instance_index].append(len(tuples))
+                tuples.append(instance_request)
+
+        # run the requests
+        results = self._run_loglikelihood(tuples, model, tokenizer, batch_size)
+
+        # TODO figure out why this is being suppressed
```

Comment:
@dirkgr I'm still looking into this one, but let me know if you have any intuition why a `warnings.warn('...')` might be suppressed within Catwalk?

When I debug into this I observe the code reaching this warning (I can even run the warning manually in the debugger terminal) and yet nothing makes it to stderr. If I put a warning in `catwalk.__main__.py` however this does work. Perhaps I'm just missing something obvious, as I haven't done a whole lot with warnings before.

Reply:
Can you trace into `initialize_logging()` and check what it sets up there? There is a call at the bottom `logging.captureWarnings(True)`, which should redirect all warnings to the logging system. Maybe it gets filtered by something?